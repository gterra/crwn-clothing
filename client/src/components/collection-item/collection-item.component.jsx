import React from "react";
import { connect } from "react-redux";
import { addCartItem } from "./../../redux/cart/cart.actions";

import {
  ImageDiv,
  AddButton,
  CollectionItemDiv,
  CollectionFooterDiv,
  NameSpan,
  PriceSpan,
} from "./collection-item.styles";

export const CollectionItem = ({ item, addCartItem }) => {
  const { imageUrl, name, price } = item;
  return (
    <CollectionItemDiv>
      <ImageDiv imageUrl={imageUrl} />
      <CollectionFooterDiv>
        <NameSpan>{name}</NameSpan>
        <PriceSpan>{price}</PriceSpan>
      </CollectionFooterDiv>
      <AddButton inverted onClick={() => addCartItem(item)}>
        Add to cart
      </AddButton>
    </CollectionItemDiv>
  );
};

const mapDispatchToProps = (dispach) => ({
  addCartItem: (item) => dispach(addCartItem(item)),
});

export default connect(null, mapDispatchToProps)(CollectionItem);
