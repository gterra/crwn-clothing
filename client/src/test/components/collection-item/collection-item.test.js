import React from "react";
import { shallow } from "enzyme";

import { CollectionItem } from "../../../components/collection-item/collection-item.component";

describe("CollectionItem component", () => {
  let wrapper;
  let mockAddItem;
  const imageUrl = "www.testImage.com";
  const mockName = "black hat";
  const mockPrice = 10;

  beforeEach(() => {
    mockAddItem = jest.fn();

    const mockProps = {
      item: {
        imageUrl: imageUrl,
        price: mockPrice,
        name: mockName,
      },
      addCartItem: mockAddItem,
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it("should render CollectionItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call addItem when AddButton clicked", () => {
    wrapper.find("AddButton").simulate("click");

    expect(mockAddItem).toHaveBeenCalled();
  });

  it("should render imageUrl as a prop on ImageDiv", () => {
    expect(wrapper.find("ImageDiv").prop("imageUrl")).toBe(imageUrl);
  });

  it("should render name prop in NameSpan", () => {
    expect(wrapper.find("NameSpan").text()).toBe(mockName);
  });

  it("should render price prop in PriceSpan", () => {
    const price = parseInt(wrapper.find("PriceSpan").text());
    expect(price).toBe(mockPrice);
  });
});
