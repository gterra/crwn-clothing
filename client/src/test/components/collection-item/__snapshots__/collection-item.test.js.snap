// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should render CollectionItem component 1`] = `
<styled.div>
  <ImageDiv
    imageUrl="www.testImage.com"
  />
  <styled.div>
    <NameSpan>
      black hat
    </NameSpan>
    <PriceSpan>
      10
    </PriceSpan>
  </styled.div>
  <AddButton
    inverted={true}
    onClick={[Function]}
  >
    Add to cart
  </AddButton>
</styled.div>
`;
